x/evm/keeper/chainKeeper.go

```diff
@@ -411,6 +412,20 @@ func (k chainKeeper) GetDeposit(ctx sdk.Context, txID types.Hash, burnAddr types
 	return types.ERC20Deposit{}, 0, false
 }
 
+// GetDeposit retrieves a confirmed/burned deposit
+func (k chainKeeper) GetDeposit(ctx sdk.Context, txID types.Hash, logIndex uint64) (types.ERC20Deposit, types.DepositStatus, bool) {
+	var deposit types.ERC20Deposit
+
+	if k.getStore(ctx, k.chainLowerKey).GetNew(key.FromStr(confirmedDepositPrefix).Append(key.FromStr(txID.Hex())).Append(key.FromUInt(logIndex)), &deposit) {
+		return deposit, types.DepositStatus_Confirmed, true
+	}
+	if k.getStore(ctx, k.chainLowerKey).GetNew(key.FromStr(burnedDepositPrefix).Append(key.FromStr(txID.Hex())).Append(key.FromUInt(logIndex)), &deposit) {
+		return deposit, types.DepositStatus_Burned, true
+	}
```

Comment:
is it not dangerous to use the same prefix again?

Reply:
I can't think of any cuz the keys before and after are built differently. Anything you noticed?

Reply:
could we get into a case where we have both types for the same tx? Could that get us into trouble while iterating?

Reply:
that can't happen due to how we implemented this at https://github.com/axelarnetwork/axelar-core/blob/bc2b06e40fd59fd20a9042ceb0305d817a060ee5/x/evm/keeper/msg_server.go#L499. To change the status, we delete first and then create a new one with a different status. I am not a fan of how it is done but at the same time I don't feel like changing it now.